### PR TITLE
P20-398: Create unit tests for curriculum content i18n sync-out

### PR DESCRIPTION
**Pull request by artem-vavilov**
1. Refactor the Dashboard `curriculum_content` resource `sync-in` business logic
2. Updated the Dashboard `curriculum_content` resource `sync-in` unit tests
3. Modularized the Dashboard `curriculum_content` resource `sync-out`
4. Created unit tests for the Dashboard `curriculum_content` resource `sync-out`
5. Created executable script `bundle exec bin/i18n/resources/dashboard/curriculum_content/sync_in.rb`
6. Created executable script `bundle exec bin/i18n/resources/dashboard/curriculum_content/sync_out.rb`
7. Refactored `ScriptConstants.i18n?`

## Links
- jira ticket: [P20-398](https://codedotorg.atlassian.net/browse/P20-398)

### CODE REVIEW

File: bin/test/i18n/test_i18n_script_utils.rb [hunk 1]
```diff
@@ -359,6 +359,28 @@ def around
...
   end
 
+  describe '.write_json_file' do
+    let(:write_json_file) {I18nScriptUtils.write_json_file(file_path, data)}
+
+    let(:file_path) {'/expected/file.txt'}
+    let(:data) {{key: {key2: 'val'}}}
+
+    it 'writes pretty json content to the file' do
+      expected_file_content = <<-JSON.strip.gsub(/^ {8}/, '')
```
Comment (mgc1194) [inline bin/test/i18n/test_i18n_script_utils.rb]: I've seen this `strip. gsub` a few times now, is it supposed to remove 8 white spaces at the beginning?

Comment (artem-vavilov) [reply bin/test/i18n/test_i18n_script_utils.rb:369]: At the beginning of each new line

Comment (artem-vavilov) [reply bin/test/i18n/test_i18n_script_utils.rb]: At the beginning of each new line, but it's better to use `<<~` heredoc syntax instead

File: bin/test/i18n/resources/dashboard/curriculum_content/test_sync_in.rb [hunk 2]
```diff
@@ -1,163 +1,260 @@
 require_relative '../../../../test_helper'
 require_relative '../../../../../i18n/resources/dashboard/curriculum_content/sync_in'
 
-class I18n::Resources::Dashboard::CurriculumContent::SyncInTest < Minitest::Test
-  def test_performing
```
Comment (mgc1194) [inline bin/test/i18n/resources/dashboard/curriculum_content/test_sync_in.rb:5]: All these changes are just switching from Unit Test to Specs syntax.

**Review by mgc1194 — APPROVED** [review]
I ran the sync and I did not notice anything off.

**Pull request merged**